### ISSUE TITLE
The result is not good as expected

### ISSUE THREAD

**Issue by HoTuanLong** [kaipackhaeuser/CXR-Patient-ReID#1]
Hi, thank for the source code.
I tried reproducing Patient Retrieval and found that my results were not as good as yours. I ran with the same setting like you introduce in README.

But this is my result:
<img width="1072" alt="image" src="https://user-images.githubusercontent.com/48116985/203929116-966aecaa-3642-479c-a5bb-7985e4631594.png">

Could you check the code version you pushed and help me ?

**Comment by kaipackhaeuser** [comment]
Dear Ho Tuan Long,

thanks for your interest in this project and for experimenting with our published code. Your attached plots do indeed look unsatisfactory as the evaluation metrics decrease with an increasing number of epochs. In theory, it should be the other way around.

I just investigated the patient retrieval setting again and cannot see any issues with our code, even after performing the training runs multiple times. Due to time constraints, I trained phase 1 for 15 epochs and phase 2 for 25 epochs on an NVIDIA Quadro RTX 8000 with the settings described in our config file. The training procedure behaves as expected and we achieve a final mAP@R of 0.94, an R-precision of 0.95, and a precision@1 of 0.98 on the test set. These values are comparable to the ones we report in our paper. Please find attached the progress of those three metrics over time (phase 2):

![plots](https://user-images.githubusercontent.com/55441069/208124563-0017b278-7079-49ef-aecb-83cad4df8a29.png)

As you mentioned in one of your previous e-mails, you are using a multi-GPU setup (4 RTX 3090 in your case). However, we developed our code for usage on a single GPU only and therefore cannot guarantee functionality in a multi-GPU setup. Our guess is that this may be causing your issue. 

Best regards,
Kai

**Comment by HoTuanLong** [comment]
Thank you for your reply by email. My issue was solved when I configured the environment exactly according to your requirement.txt file. I think my cause is when I try to run code on multiple GPUs and use different versions of Pytorch Lightning.
